Add a brief style guide

bot190 opened the pull request:
I propose that we follow the Google Java Style Guide, with some small modifications. Namely we should use 4 spaces instead of the propsed 2. I've added a style.md that includes the important points as well as some examples.

Is there anything else we should include in the style guide? I believe for most things that aren't covered here the linked Google Java Style Guide provides adequate instruction.

bot190 committed:
Add a brief style guide

I propose that we follow the Google Java Style Guide, with some small modifications. Namely we should use 4 spaces instead of the propsed 2. I've added a style.md that includes the important points as well as some examples.

bot190 committed:
Use spaces instead of tabs

Replace all tabs with 4 spaces to match the style guide.

ghost commented:
Only other thing I can think of (that came up the other day) was order of elements within a class: declaration, properties, then fields (or whatever they are called for Java), constructors, methods, and finally commands. It tends to be ugly, but it might not be a bad idea to force using headers to denote each section (think regions, from C#)? I saw the other day that someone had those different elements all mixed up in their drive class, and it made it hard to compare between different people's code.

msoucy commented:
Agreed, add something about the order of members

bot190 committed:
Add section on ordering

Add a section describing the order of java files. This makes it easier for everyone to quickly look at subsystems.

bot190 commented:
I added a section on ordering of members. I'm also converting all the files to using spaces for indentation instead of tabs. We'll want to make sure everyone pulls this before they do any more development.
